src/shogun/statistical_testing/internals/Block.cpp

```diff
@@ -48,7 +48,9 @@ Block::Block(CFeatures* feats, index_t index, index_t size) : m_feats(feats)
 
 	// create a shallow copy and subset current block separately
 	CFeatures* block=FeaturesUtil::create_shallow_copy(feats);
+#ifdef USE_REFERENCE_COUNTING
 	ASSERT(block->ref_count()==0);
```

Comment:
So my thoughts are

 * if this assertion could possibly ever fail, then there is some serious problem in the `create_shallow_copy` above.
 * If this assertion never fails, i.e. the above line guarantees something that is documented in its interface and maybe even unit tested, then why put it?